pennylane_lightning/src/bindings/Bindings.cpp

```diff
@@ -12,658 +12,253 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <cassert>
 #include <set>
 #include <tuple>
 #include <vector>
 
+#include <iostream>
+
 #include "AdjointDiff.hpp"
+#include "IndicesUtil.hpp"
 #include "JacobianProd.hpp"
-#include "StateVector.hpp"
+#include "StateVectorBase.hpp"
+#include "StateVectorRaw.hpp"
 #include "pybind11/complex.h"
 #include "pybind11/numpy.h"
 #include "pybind11/pybind11.h"
 #include "pybind11/stl.h"
+#include "pybind11/stl_bind.h"
 
 /// @cond DEV
 namespace {
 using namespace Pennylane::Algorithms;
-using Pennylane::StateVector;
+using Pennylane::StateVectorBase;
+using Pennylane::StateVectorRaw;
+
+using Pennylane::Internal::callGateOps;
+using Pennylane::Internal::implementedGatesForKernel;
+
+using Pennylane::Internal::GateOpsFuncPtrPairs;
+
 using std::complex;
 using std::set;
 using std::string;
 using std::vector;
+} // namespace
+/// @endcond
 
 namespace py = pybind11;
```

Comment:
I wonder if we should move this to a `Bindings.hpp` now, given the large number of `include` and `using` operations. What do you think?

Reply:
Agree on this.

Reply:
Now we have `Bindings.hpp`

Reply:
https://github.com/PennyLaneAI/pennylane-lightning/blob/c080a2aa3285b3f9113ab803c2baabeca48b655f/pennylane_lightning/src/bindings/Bindings.hpp